packages/jest/src/matchers.tsx

```diff
@@ -56,9 +57,14 @@ export function toHaveCompiledCss(
       });
     }
 
-    if (containsClassNames(classNames, css)) {
-      foundStyles.push(...stylesToFind.filter(styleToFind => css.includes(styleToFind)));
-    }
+    classNames.forEach(c => {
+      let matcher = c;
+      if (state) matcher += `:${state}`;
+      if (media) matcher = `${media}{.*${matcher}`;
+      if (css.match(matcher)) {
```

Comment:
this means if there is no matcher defined it will just mark all styles it finds as assertable yeah? 

what's your thoughts on this? i can see two sides to it 

- it might give false positives so people think something is working when it isnt
- we treat matchers as narrowing and its ok that the default case asserts on every state

Reply:
Thanks for reviewing. I think a better approach would be to not do a text search and to convert this into an AST and then work with rules. This is done both in emotion-jest and styled-component jest matchers. 

Also, at least in `jest-styled-components` matchers are mandatory and it doesn't work without narrowing, which I like too! 

Let me know your thoughts and I can give it a spin.

Reply:
i like it - let's give it a spin